eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -0,0 +1,147 @@
+jobs:
+  - job: 'Build'
+    variables:
+    - template: ../variables/globals.yml
+    pool:
+      vmImage: 'ubuntu-16.04'
+
+    steps:
+      - task: NodeTool@0
+        inputs:
+          versionSpec: '$(NodeVersion)'
+        displayName: 'Install Node.js $(NodeVersion)'
+
+      - script: |
+          node common/scripts/install-run-rush.js install
+        displayName: 'Install dependencies'
+
+      - script: |
+          node common/scripts/install-run-rush.js build --to $(PackageName)
+        displayName: 'Build libraries'
+
+      - script: |
+          node common/scripts/install-run-rush.js pack --to $(PackageName)
+        displayName: 'Pack libraries'
+
+      - task: CopyFiles@2
+        inputs:
+          contents: 'sdk/**/**/*.tgz'
+          targetFolder: $(Build.ArtifactStagingDirectory)
+          flattenFolders: true
+        displayName: 'Copy packages'
+
+      - task: PublishPipelineArtifact@0
```

Comment:
Will these definitions also be used to queue builds manually for release? It seems like they should support that. If so, maybe we can take this opportunity to fix something related to the release process.

Right now, `--to packagename` will build `packagename` and all of its dependency projects. This means that the resulting artifact "packages" will potentially contain multiple tarballs. This causes a problem because the release pipeline will attempt to download and publish any tarballs from the "packages" artifact, when you only really want to publish the single named package tarball.

In addition to copying all tarballs to the "packages" artifact, we should also use the supplied `PackageName` to copy only that single tarball intended for publishing into a different "publish" artifact. Then we can just update the release pipeline to look at the "publish" artifact when downloading tarballs to publish.

Reply:
I would kind of expect if we do a filtered build it should only package the whatever we filter the build down to. Is there a reason that cannot be the behavior?

Reply:
The YML file that @bsiegel references here probably should be able to be used by the PR validation pipelines as well as the full build runs. I'm focusing on the PR validation builds first and then I'll come back and tweak for the full build runs. There are a couple of different things we could do to omit/include the ```--to``` text depending on whether PackageName is present.

We probably also only want to build modules which are part of the data plane space, is there a Rush command that can just target that subset?

Reply:
Currently only data plane libraries are included in the Rush workspace. "groups" is not yet a feature in Rush, but I submitted a proposal for such. At some point when I have time I might work on contributing that feature.

Reply:
Also, @mitchdenny mentioned that template expressions should work, even outside of template files. Template expressions allow the construct `- ${{ if condition() }}` to conditionally execute part of the YML. That was totally non-obvious to me.

Reply:
OK - I've switched over using template expressions and parameters for this scenario. Could have used variables with conditional logic but the benefit here is that the job that runs doesn't contain skipped steps which makes it easier on folks I believe. I think generally speaking where I've used variables previously in the PR validation pipelines that I'll switch to using parameters for passing information down into templates. It is a grey area though as to which one is best.

Variables are mutable during the build (and at queue time) - so it just depends on the scenario.